### PR TITLE
Fix broken TGrains `amp` handling

### LINKED ISSUE #2807: TGrains amp argument has no effect
It seems the TGrains Ugen's amp argument has no effect on the amplitude of the grains. I'm running this on SC 3.8.0 on a macbook pro (Yosemite). 

Here's some code to illustrate:

```
b = Buffer.read(s, Platform.resourceDir +/+ "sounds/a11wlk01.wav");
{ // This should fade in... but starts at full volume
	TGrains.ar(2, Impulse.ar(15), b, centerPos: Line.ar(0, 5, 10, doneAction: 2), amp: Line.ar(0, 1, 10));
}.play;
```

### PR DESCRIPTION
**Pull request by brianlheim**
Fixes #2807, a regression caused by #2136 (https://github.com/supercollider/supercollider/pull/2136/commits/f35a3f7f1552266fda04e326474dd678177b2ba5).

This should work correctly now:
```supercollider
b = Buffer.read(s, Platform.resourceDir +/+ "sounds/a11wlk01.wav");
{ // This should fade in... but starts at full volume
	TGrains.ar(2, Impulse.ar(15), b, centerPos: Line.ar(0, 5, 10, doneAction: 2), amp: Line.ar(0, 1, 10));
}.play;
```

### CODE REVIEW

**Comment by tiagmoraismorgado** [comment]
**some suggestions** _(sent from a library computer, may not compile due to an eventual error. i am sorry if that's the case)_

```

~myBuffer = Buffer.read(s, Platform.resourceDir +/+ "sounds/a11wlk01.wav");


{ // This should fade in... but starts at full volume
	TGrains.ar(~myBuffer.numChannels, Impulse.kr(200), ~myBuffer.bufnum, 
                   centerPos: Line.ar(0, 5, 10, doneAction: 2) + TRand.kr(0, 0.01, Impulse.kr(400)), 
                   pan: WhiteNoise.kr(0.6), amp: Line.ar(0, 1, 10), interp: 4);
}.scope(s);


```

_

> ever tried ever failed no matter try again fail again fail better

_

**Comment by brianlheim** [comment]
@tiagmoraismorgado Make sure you understand the PR before commenting please. This is just about making sure the `amp` parameter to TGrains functions correctly.

File: server/plugins/DelayUGens.cpp [hunk 1]
```diff
@@ -6216,8 +6216,8 @@ void TGrains_next(TGrains *unit, int inNumSamples)
 					pan = sc_clip(pan * 0.5f + 0.5f, 0.f, 1.f);
 					panangle = pan * pi2_f;
 				}
-				pan1 = grain->pan1 = cos(panangle);
-				pan2 = grain->pan2 = sin(panangle);
+				pan1 = grain->pan1 = amp * cos(panangle);
+				pan2 = grain->pan2 = amp * sin(panangle);
 			} else {
 				grain->chan = 0;
 				pan1 = grain->pan1 = 1.;
```
Comment (snappizz) [inline server/plugins/DelayUGens.cpp]: You'll want to multiply by `amp` in the one-channel case too.

Comment (brianlheim) [inline server/plugins/DelayUGens.cpp]: Done!

**Pull request merged**